src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/CollectionProperty.java

```diff
@@ -0,0 +1,123 @@
+package io.stargate.sgv2.jsonapi.service.bridge.executor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.sgv2.jsonapi.config.constants.DocumentConstants;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import java.util.Optional;
+
+/**
+ * Refactored as seperate class that represent a collection property.
+ *
+ * @param collectionName
+ * @param vectorEnabled
+ * @param vectorSize
+ * @param similarityFunction
+ * @param vectorizeServiceName
+ * @param modelName
+ */
+public record CollectionProperty(
```

Comment:
Refactored this as separate class so that it can be used from the create collection command.

Reply:
Nice!

Minor nit: name itself could probably be improved a bit. Not sure if `CollectionSettings` would be a better name? (CollectionConfig possibly, but we use `-Config` for things configured with configuration files and this is more dynamic so `-Settings` seems better).

Reply:
Done.